iris_grib/message.py

```diff
@@ -404,12 +412,22 @@ def __getitem__(self, key):
         if key not in self._cache:
             if key == 'numberOfSection':
                 value = self._number
-            elif key not in self._keys:
-                raise KeyError('{!r} not defined in section {}'.format(
-                    key, self._number))
             else:
+                keys = [key]
+                # Also check known alias alternatives
+                if key in KEY_ALIAS:
+                    keys.append(KEY_ALIAS[key])
+                keys = np.array(keys)
+                found = [k in self._keys for k in keys]
+                if not any(found):
+                    emsg = f"{key} not defined in section {self._number}"
+                    raise KeyError(emsg)
+                # Take the first valid key, original or alias.
+                key = keys[found][0]
```

Comment:
I'm sure this works but, given that we only support a _single_ alias for any key, I'm finding this even-handed "there are a list of possibles" approach a bit obscure.
Can't we just "if then if else fail" it ?

```suggestion
                if key not in self._keys:
                    key2 = KEY_ALIAS.get(key)
                    if key2 and key2 in self._keys:
                        key = key2
                    else:
                        emsg = f"{key} not defined in section {self._number}"
                        raise KeyError(emsg)
```

Not promising this definitely works (!)